### PR TITLE
SCP XInput Bridge Integration

### PR DESCRIPTION
**Pull request by CookiePLMonster**
### Description of Changes
DsHidMini, the newest driver for DualShock 3 gamepads, comes with a [SCP XInput Bridge DLL](https://vigem.org/projects/DsHidMini/SCP-XInput-Bridge/), implementing a SCP-era extension of the XInput API. With this extension, button pressure of DS3 gamepads may be read by a supported application. PCSX2 used to support this API extension during lilypad days (AFAIK), but now it was gone. This PR brings it back as a part of the XInput backend, and kicks in automatically once DsHidMini's `xinput1_3.dll` is placed in the PCSX2 directory.

This PR also fixes wrong triggers scaling for "normal" XInput.

### Rationale behind Changes
Pressure sensitivity is sometimes important, and SCP XInput Bridge is likely the cleanest way to obtain it. Plus, it used to be supported.

### Suggested Testing Steps
1. Dust off, buy, steal and/or 3D print a DualShock 3.
2. Install [DsHidMini](https://vigem.org/projects/DsHidMini/How-to-Install/).
3. Put DsHidMini's `xinput1_3.dll` in PCSX2 directory.
4. Launch PCSX2 and toggle Enable XInput Source. Ensure the gamepad is mapped properly in XInput mode.
5. Launch any application that you can verify pressure sensitive input data, e.g. racing games or `padtest`.
6. Verify that pressure sensitivity works as intended.

pressure

### CODE REVIEW

File: pcsx2/Frontend/XInputSource.cpp [hunk 1]
```diff
@@ -158,29 +165,44 @@ void XInputSource::Shutdown()
...
 void XInputSource::PollEvents()
 {
 	for (u32 i = 0; i < NUM_CONTROLLERS; i++)
 	{
-		XINPUT_STATE new_state;
-		const DWORD result = m_xinput_get_state(i, &new_state);
 		const bool was_connected = m_controllers[i].connected;
+
+		SCP_EXTN new_state_scp;
+		DWORD result = m_xinput_get_extended != nullptr ? m_xinput_get_extended(i, &new_state_scp) : ERROR_NOT_SUPPORTED;
```
Comment (stenzek) [inline pcsx2/Frontend/XInputSource.cpp:178]: Nitpick: `m_xinput_get_extended ? m_xinput_get_extended(...) : ...`

File: pcsx2/Frontend/XInputSource.h [hunk 2]
```diff
@@ -21,6 +21,39 @@
 #include <mutex>
 #include <vector>
 
+// SCP XInput extension
+typedef struct
```
Comment (stenzek) [inline pcsx2/Frontend/XInputSource.h:25]: This could be forward declared in the header, with the actual struct in the cpp file.

**Review by stenzek — APPROVED** [review]
Otherwise LGTM

**Comment by nefarius** [comment]
Related https://github.com/ViGEm/DsHidMini/issues/213

**Pull request merged**